### PR TITLE
-profile page

### PR DESCRIPTION
**Pull request by SudevSanthosh**
-react custom hooks
-custom functions

### CODE REVIEW

**Review by dheeraj-murali — COMMENTED** [review]
This looks really good.
useGetApiData hook is on point, good abstraction and reusable as well.

Can you see if you can combine getInitials, getUserAddress and bring it to getUserDetails. and maybe use a switch conditions to get the properties

File: src/hooks/useGetApiData.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const useGetApiData = (url) => {
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => setData(data));
+  }, [url]);
+
+  return [data, setData];
```
Comment (dheeraj-murali) [inline src/hooks/useGetApiData.js]: It would be better not to return setData here. we wouldn't want the ability tp update the response data.

**Review by dheeraj-murali — CHANGES_REQUESTED** [review]

File: src/hooks/useGetApiData.js (hunk 1 above)
Comment (SudevSanthosh) [reply src/hooks/useGetApiData.js]: resolved

**Comment by SudevSanthosh** [comment]
used switch conditions to get the properties of user

**Pull request merged**